### PR TITLE
Move kind development instructions to top of dev cluster instructions.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

kind provides an easy way for people to get started with developing
Tekton. This change promotes kind to the top of the cluster getting
started instructions, and simplifies the setup with easy copy+paste
commands.

Keeps reference to tekton_in_kind.sh, but does not rely on it for the
core setup instructions since it installs multiple tekton components
besides pipelines.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -256,6 +256,27 @@ The recommended minimum development configuration is:
   - 8 GB of (actual or virtualized) platform memory
 - Node autoscaling, up to 3 nodes
 
+#### Using [KinD](https://kind.sigs.k8s.io/)
+
+[KinD](https://kind.sigs.k8s.io/) is a great tool for working with Kubernetes clusters locally. It is particularly useful to quickly test code against different cluster [configurations](https://kind.sigs.k8s.io/docs/user/quick-start/#advanced).
+
+1. Install [required tools](./DEVELOPMENT.md#install-tools) (note: may require a newer version of Go).
+2. Install [Docker](https://www.docker.com/get-started).
```
Comment (vdemeester) [inline DEVELOPMENT.md:264]: nit: or anything supported by `kind` (like `podman`)

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -256,6 +256,27 @@ The recommended minimum development configuration is:
...
+   
+   ```sh
+   $ kind create cluster
+   ```
+
+4. Configure [ko](https://kind.sigs.k8s.io/):
+   
+   ```sh
+   $ export KO_DOCKER_REPO="kind.local"
+   $ export KIND_CLUSTER_NAME="kind"  # only needed if you used a custom name in the previous step
```
Comment (vdemeester) [inline DEVELOPMENT.md:275]: I don't remember, if the cluster name is different, shouldn't `KO_DOCKER_REPO` be different too ?

**Review by vdemeester — APPROVED** [review]

File: DEVELOPMENT.md (hunk 2 above)
Comment (wlynch) [reply DEVELOPMENT.md:275]: Nope! Only KIND_CLUSTER_NAME needs to be set. 🧙 

I use this all the time having separate kind clusters for vanilla pipelines and pipelines + results.

File: DEVELOPMENT.md (hunk 1 above)
Comment (wlynch) [reply DEVELOPMENT.md:264]: kind unfortunately doesn't have a good page to link to at the moment for supported runtimes, so I'm a bit hesitant to include this right now (I want to make the getting started instructions as simple as possible with minimal research needed).

Comment (vdemeester) [reply DEVELOPMENT.md:264]: yeah make sense 👍🏼

**Comment by sbwsg** [comment]
Nice, cheers!

/lgtm

**Pull request merged**